### PR TITLE
#7 Backport to API 14+

### PR DESCRIPTION
**Pull request by Mayankmkh**
Uses [Transitions-Everywhere](https://github.com/andkulikov/Transitions-Everywhere) library for backward compatibility.

### CODE REVIEW

**Comment by JoaquimLey** [comment]
@Mayankmkh thank you for your submission, I've been working on the same issue 👍 
Do you have an idea do backport without having to create a dependency?
Since this is a library we wanna have as little impact as possible, and importing a whole another library into this one doesn't seem the right way. Depending on the License it may even be possible to copy the actual class (once again, we must respect the creator's license).

**Comment by Mayankmkh** [comment]
Hi @JoaquimLey I found no other way to backport this library without using these dependencies. As the vector-compat library is under apache license v2.0, so it is possible to use code from that. Transitions-everywhere doesn't have license file so the developer needs to be contacted.
I'll be happy to know any other way to do so.

**Comment by Mayankmkh** [comment]
@JoaquimLey Another way could be to use AnimationDrawable instead of animated vector drawable for fab drawable morphing.

**Comment by JoaquimLey** [comment]
The framework does provide us the tools required to backport, of course some animations will be lost, including the VectorDrawable.

This is just a case of seeing the provided methods and work our way around it. I've pushed this [branch](https://github.com/JoaquimLey/faboptions/tree/feature-backPort) with an initial draft/ideas to backport, it can be a good foundation.

**Comment by JoaquimLey** [comment]
Like I said, it's not optimal to be importing 3rd party libraries to this one, this will greatly increase the method count and general overhead. If these libraries are able, we sure can too. Are you interested in pursuing a more "clean" solution?
Thanks in advance for your interest 👍

**Comment by Mayankmkh** [comment]
@JoaquimLey Thanks for your advice. As you suggested I would definitely try to come up with a better approach to minimise the library footprint and refrain from using 3rd party libs.

**Comment by JoaquimLey** [comment]
Looking forward for your solution @Mayankmkh

**Comment by Mayankmkh** [comment]
@JoaquimLey Sorry, it is currently not possible for me to work for 2 weeks due to time issues. I would like to contribute after that. I hope you understand.

**Comment by JoaquimLey** [comment]
No problem, I'll be looking for it 👍

**Comment by Mayankmkh** [comment]
Hi @JoaquimLey I have removed third party libraries and tried to backport the animations to API 14. Please review the commits.

File: faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java [hunk 1]
```diff
@@ -189,6 +192,24 @@ private void animateButtons(boolean isOpen) {
         }
     }
 
+    private int fetchAccentColor() {
```
Comment (JoaquimLey) [inline faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java]: Is this to enable colorize the icons?

**Comment by JoaquimLey** [comment]
Rebase with current developement please @Mayankmkh

File: faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java [hunk 2]
```diff
@@ -91,7 +95,7 @@ private void inflateButtonsFromAttrs(Context context, AttributeSet attrs) {
     }
 
     public void setButtonsMenu(Context context, @MenuRes int menuId) {
-        mMenu = new MenuBuilder(context);
+        Menu mMenu = new MenuBuilder(context); // TODO: 22/11/2016 add items in runtime
```
Comment (Mayankmkh) [inline faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java]: I just converted mMenu from global object to local object. TODO was already there.

File: faboptions/build.gradle [hunk 3]
```diff
@@ -44,6 +44,6 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile "com.android.support:design:$DESIGN_LIBRARY_VERSION"
     compile "com.andkulikov:transitionseverywhere:1.7.0"
-
+    compile 'com.wnafee:vector-compat:1.0.5'
```
Comment (Mayankmkh) [inline faboptions/build.gradle]: Sorry for this. I forgot to remove these libraries from gradle.

**Comment by Mayankmkh** [comment]
@JoaquimLey Thanks for your comments. I will rebase it with development branch and update you soon

File: faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java (hunk 2 above)
Comment (JoaquimLey) [inline faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java]: The idea is to leave it as global variable so we can add items at run time, but no worries.

File: faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java [hunk 4]
```diff
@@ -16,47 +16,49 @@
...
     private boolean mIsOpen;
     private View.OnClickListener mListener;
 
-    private Menu mMenu; // TODO: 22/11/2016 add items in runtime
     private FloatingActionButton mFab;
 
     private View mBackground;
     private View mSeparator;
     private FabOptionsButtonContainer mButtonContainer;
+    private int animDuration;
```
Comment (JoaquimLey) [inline faboptions/src/main/java/com/joaquimley/faboptions/FabOptions.java]: Please change this to hungarian.

**Comment by JoaquimLey** [comment]
Do you want me to fix the conflict?

**Comment by Mayankmkh** [comment]
I have already fixed the conflicts.

**Pull request merged**